src/ScottPlot4/ScottPlot.Demo/ScottPlot.Demo.Avalonia/CookbookWindow.axaml.cs

```diff
@@ -14,14 +14,14 @@ public CookbookWindow()
 
         private void DemoSelected(object sender, SelectionChangedEventArgs e)
         {
-            TreeView DemoTreeview = (TreeView)sender; ;
+            TreeView DemoTreeview = (TreeView)sender;
 
             Cookbook.TreeNode selectedDemoItem = null;
             if (DemoTreeview.SelectedItems.Count > 0)
             {
                 selectedDemoItem = (Cookbook.TreeNode)DemoTreeview.SelectedItems[0];
             }
-            if (selectedDemoItem != null && selectedDemoItem.ID != null)
+            if (selectedDemoItem?.ID != null)
```

Comment:
Not sure what you mean here, but the two statements should work exactle the same way. Could you please give me some more details about your thoughts here?

Reply:
In general for a null check you're supposed to prefer `is null` or `is not null` rather than `== null` or `!= null`. We've been inconsistent about this throughout the codebase but this is what Microsoft and most other style guides recommend.

`==` is affected by operator overloading which is rarely the behaviour that one wants for a null check. An operator overload that changes comparison with null should be rare, but it also allows more helpful warnings and errors (for example it can prevent null checks on non-nullable value types).

And there are potentially performance differences, `is` can always be compiled to MSIL's `ceq` instruction, where `==` is only compiled that way if `==` isn't overloaded (and you're using a sufficiently recent version of Roslyn).

Reply:
Got it, I was not fully aware of it. Thanks for the explanation. I've fixed it.